### ISSUE TITLE
LLaMAModel._causal_lm_process中的labels和logits对齐方法疑问

### ISSUE THREAD

**Issue by chivychao** [alibaba/Megatron-LLaMA#53]
`# [invalid] Shift so that tokens < n predict n`
`# Do not need to shift here`
`shift_logits = logits[..., :-1, :].contiguous()`
`shift_labels = labels[..., :-1].contiguous()`

其中还有必要把最后一位给抹掉吗？是不是直接可以取整个序列呢？

**Comment by li-yi-dong** [comment]
如果用的是Megatron-LM 原生的dataset 可以不用移位。具体要看label 和样本是怎么对应的。

**Comment by chivychao** [comment]
那是否可以不使用self._causal_lm_process()来计算loss，仍然使用post_language_model_processing()来计算loss呢？

**Comment by li-yi-dong** [comment]
> 那是否可以不使用self._causal_lm_process()来计算loss，仍然使用post_language_model_processing()来计算loss呢？

可以，但我们发现这个实现和HuggingFace 上的实现计算结果有些差异，需要你自己评估。

**Comment by chivychao** [comment]
我将最后层的结构从LLaMAModel.lm_head更改为TransforerLanguageModel.output_layer，然后重新启用post_language_model_processing()来计算loss后发现，pipeline并行的最后一组卡的显存占用仍然要高出前面的几组卡一大截，请问这是正常现象吗？
